capsules/extra/src/ieee802154/driver.rs

```diff
@@ -1143,7 +1143,12 @@ impl<'a, M: device::MacDevice<'a>> device::RxClient for RadioDriver<'a, M> {
                         // the data offset, the data length, and the MIC length.
                         rbuf[(offset + USER_FRAME_METADATA_SIZE)..(offset + user_frame_len)]
                             .copy_from_slice(&buf[PSDU_OFFSET..frame_len]);
-                        rbuf[offset].set(data_offset as u8);
+
+                        // data_offset includes the PSDU offset, so we must subtract this value and add the
+                        // length of the metadata to obtain the data offset for the user frame.
```

Comment:
This comment is really vague and I don't understand it. What does it mean for `data_offset` to "include" the PSDU offset. The PSDU (PHY Service Data Unit?) is meaningless at this level (this is a client of the Mac device, which has already stripped the PSDU information?

I.e. `data_offset` (which is documented as the first byte in the user frame format, but not clearly defined anywhere in this file what it means) probably should semantically point to the offset where the MAC payload starts.

What is the meaning of this new value that is for some reason?